CMakeLists.txt

```diff
@@ -35,6 +35,9 @@ macro(build_yaml_cpp)
   endif()
 
   if(DEFINED CMAKE_TOOLCHAIN_FILE)
+    if(QNX)
+      set_qnx_external_project_options()
+    endif()
```

Comment:
I'm definitely not a CMake expert, but this seems pretty magical to me.  Since there are no `find_package` or `include` related to this, it just seems to come from nowhere.  Is there any way to make it more obvious where this is coming from?

Reply:
This macro is provided as part of the qnx toolchain file. It is needed for all vendor packages in ros2. https://github.com/asobhy-qnx/ros2/blob/master/platform/qnx.nto.toolchain.cmake

Including the toolchain file when building using the -DCMAKE_TOOLCHAIN _FILE option will automatically make this macro available to all the packages being built. Also the toolchain file sets the variable QNX so if it is true then the macro “set_qnx_external_project_options()” must exists.

I could just include the options inline if this would make it less ambiguous.